### DISCUSSION TITLE
[openssl-3.0.12] How to use a provider when signing in curl？

### DISCUSSION THREAD

**Discussion by Czzqq** [openssl/openssl#24999] [Q&A]
Due to the confidentiality of the private key, I wish for the curl operation to call the provider I have implemented during the signing process. However, the SSL currently produces this output, and the EVP_PKEY I return indeed does not contain the private key. How should I handle this part of the process?
`* OpenSSL/3.0.12: error:020000B3:rsa routines::missing private key`

**Comment by levitte** [comment]
What operations does your provider provide implementations for?

**Reply by Czzqq** [reply]
provider_store: read a public key from TEE and construct a pkey(only have public key info ,but it's a key pair type)
provider_keymgmt: to load or new key_ctx
provider_signature: to sign and verify.
![image](https://github.com/user-attachments/assets/b16e60a8-a905-4c1b-9514-369b4c915155)
![image](https://github.com/user-attachments/assets/eea243ce-b4dd-4da2-94a1-1259d8048a4b)
![image](https://github.com/user-attachments/assets/8e5dcc1e-6903-47d2-9102-c8f8911e58ae)

Besides this， i code a openssl EVP demo is nice to process, call STORE_INFO to load a pkey , sign & verify.
![image](https://github.com/user-attachments/assets/ec1dd09b-487a-4c42-9256-fc3b3847a0d7)

**Reply by Czzqq** [reply]
I am not entirely sure about the role of a large part of the interfaces in CURL, so I have defined them all just to track which are called appropriately. It would be great if you could correct any unnecessary implementations based on my requirements.
